File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/offices-management/models/office-details.dto.ts
import {OfficeDto} from './office.dto';
import {OfficeUserDto} from './office-user.dto';
import {GUID} from '../../../../../shared/types/guid.type';

export interface OfficeDetailsDto extends OfficeDto {
  countryId: GUID;
  users: OfficeUserDto[];
}

File: src/Application.Operation/Features/Admin/Offices/DTOs/OfficeDetailsDto.cs
using Tawtheef.Application.Common.Models;

namespace Application.Operation.Features.Admin.Offices.DTOs;

public sealed record OfficeDetailsDto
{
    public Guid Id { get; init; }
    public string NameAr { get; init; } = string.Empty;
    public string NameEn { get; init; } = string.Empty;
    public Guid CountryId { get; init; }
    public List<DropdownOptions> SupportedCountries { get; init; } = [];
    public bool IsActive { get; init; }
    public string AdminNameAr { get; init; } = string.Empty;
    public string AdminNameEn { get; init; } = string.Empty;
    public string AdminEmail { get; init; } = string.Empty;
    public string? PhoneCountryCode { get; init; }
    public string? PhoneNumber { get; init; }
    public List<OfficeUserDto> Users { get; init; } = [];
}
